### ISSUE TITLE
providePrefixAndSuffixTextForRename only seems to work for global 'configure'

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#29585]
**TypeScript Version:**  3.3.0-rc

**Search Terms:**
- providePrefixAndSuffixTextForRename

**Bug**
In VS Code, we set provide the  `providePrefixAndSuffixTextForRename` preference on each file we configure preferences for:

```
[Trace  - 11:26:25 AM] Sending request: configure (9). Response expected: yes. Current queue length: 2
Arguments: {
    "file": "/Users/matb/projects/san/src/index.ts",
    "formatOptions": {
      ...
    },
    "preferences": {
        ...,
        "providePrefixAndSuffixTextForRename": true,
        "allowRenameOfImportPath": true
    }
}
```

However it looks like this setting is not observed if configured per-file, it only applies if we configure it for the entire project:

```
[Trace  - 11:29:28 AM] Sending request: configure (0). Response expected: no. Current queue length: 0
Arguments: {
    "hostInfo": "vscode",
    "preferences": {
        "providePrefixAndSuffixTextForRename": true
    }
}
```

/cc @uniqueiniquity

**Comment by mjbvz** [comment]
Not sure if this also applies to `allowRenameOfImportPath`

**Comment by uniqueiniquity** [comment]
It does also apply to `allowRenameOfImportPath`. Sorry for missing this.

**Comment by mjbvz** [comment]
Thanks for the quick follow up.

VS Code can easily pass this preference globally but in order to allow users  to opt out of the new rename behavior (#29238) we will need to set it per-file. So while we'd prefer a fix for this in 3.3, we could work around this
